src/Documentation/tutorials_and_samples/tutorial_1_helloworld.md

```diff
@@ -0,0 +1,377 @@
+
+```csharp
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using HelloWorld.Grains;
+using Microsoft.Extensions.Logging;
+using Orleans;
+using Orleans.Configuration;
+using Orleans.Hosting;
+
+namespace OrleansSiloHost
+{
+    public class Program
+    {
+        public static int Main(string[] args)
+        {
+            return RunMainAsync().Result;
+        }
+
+        private static async Task<int> RunMainAsync()
+        {
+            try
+            {
+                var host = await StartSilo();
+                Console.WriteLine("Press Enter to terminate...");
+                Console.ReadLine();
+
+                await host.StopAsync();
+
+                return 0;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+                return 1;
+            }
+        }
+
+        private static async Task<ISiloHost> StartSilo()
+        {
+            // define the cluster configuration
+            var builder = new SiloHostBuilder()
+                .UseLocalhostClustering()
+                .Configure<ClusterOptions>(options =>
+                {
+                    options.ClusterId = "dev";
+                    options.ServiceId = "HelloWorldApp";
+                })
+                .Configure<EndpointOptions>(options => options.AdvertisedIPAddress = IPAddress.Loopback)
+                .ConfigureApplicationParts(parts => parts.AddApplicationPart(typeof(HelloGrain).Assembly).WithReferences())
+                .ConfigureLogging(logging => logging.AddConsole());
+
+            var host = builder.Build();
+            await host.StartAsync();
+            return host;
+        }
+    }
+}
+
+
+```
+
+### OrleansClient - Program.cs
+
+```csharp
+using HelloWorld.Interfaces;
+using Orleans;
+using Orleans.Runtime;
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Orleans.Configuration;
+
+namespace OrleansClient
+{
+    /// <summary>
+    /// Orleans test silo client
+    /// </summary>
+    public class Program
+    {
+        const int initializeAttemptsBeforeFailing = 5;
+        private static int attempt = 0;
+
+        static int Main(string[] args)
+        {
+            return RunMainAsync().Result;
+        }
+
+        private static async Task<int> RunMainAsync()
+        {
+            try
+            {
+                using (var client = await StartClientWithRetries())
+                {
+                    await DoClientWork(client);
+                    Console.ReadKey();
+                }
+
+                return 0;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                Console.ReadKey();
+                return 1;
+            }
+        }
+
+        private static async Task<IClusterClient> StartClientWithRetries()
+        {
+            attempt = 0;
+            IClusterClient client;
+            client = new ClientBuilder()
```

Comment:
same here. Please keep in mind, this tutorial would be paired with HelloWorld sample, so we don't need to duplicate information already provided here. Tutorial is a good place to explain key points about the sample, providing information which wasn't there. Such as how to use ClientBuilder.

Reply:
I removed the copied code.